test/gtest/ucm/malloc_hook.cc

```diff
@@ -799,3 +800,78 @@ UCS_TEST_F(malloc_hook, bistro_patch) {
     EXPECT_NE(patched, origin);
 #endif
 }
+
+/* test for mmap events are fired from non-direct load modules
+ * we are trying to load lib1, from lib1 load lib2, and
+ * fire mmap event from lib2 */
+UCS_TEST_F(malloc_hook, dlopen) {
+    typedef void (fire_mmap_f)(void);
+    typedef void* (load_lib_f)(const char *path);
+
+    const char *libdlopen_test  = "/.libs/libdlopen_test.so";
```

Comment:
looks a bit ugly to use paths like this in gtest..
maybe do it in separate test in test_jenkins.sh (with mpi malloc hook tests)?